test/python/transpiler/test_basic_mapper.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Test the Basic Mapper pass"""
+
+import unittest
+from copy import deepcopy
+from qiskit.transpiler.passes import BasicMapper
+from qiskit.mapper import Coupling
+from qiskit.converters import circuit_to_dag
+from qiskit import QuantumRegister, QuantumCircuit
+from ..common import QiskitTestCase
+
+
+class TestBasicMapper(QiskitTestCase):
+    """ Tests the BasicMapper pass."""
+
+    def test_trivial_case(self):
+        """No need to have any swap, the CX are distance 1 to each other
+         q0:--(+)-[U]-(+)-
+               |       |
+         q1:---.-------|--
+                       |
+         q2:-----------.--
+
+         Coupling map: [1]--[0]--[2]
+        """
+        coupling = Coupling({0: [1, 2]})
+
+        qr = QuantumRegister(3, 'q')
+        circuit = QuantumCircuit(qr)
+        circuit.cx(qr[0], qr[1])
+        circuit.h(qr[0])
+        circuit.cx(qr[0], qr[2])
+
+        dag = circuit_to_dag(circuit)
+        before = deepcopy(dag)
```

Comment:
no need for a deepcopy here. just assert that `after` and `dag` are the same. building `before` is redundant.

Reply:
remove them in 66c33e06